guppylang/definition/ty.py

```diff
@@ -42,12 +42,6 @@ class OpaqueTypeDef(TypeDef, CompiledDef):
     to_hugr: Callable[[Sequence[Argument]], tys.Type]
     bound: tys.TypeBound | None = None
 
-    def __getitem__(self, item: Any) -> "OpaqueTypeDef":
-        """Dummy implementation to allow generic instantiations in type signatures that
-        are evaluated by the Python interpreter.
-        """
-        return self
```

Comment:
This is no longer needed since the user now receives a `GuppyDefinition` which already implements`__getitem__`